kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -225,4 +210,39 @@ public static void applyCommonConfiguration(Config config, HttpClient.Builder bu
     HttpClientUtils.createApplicableInterceptors(config, factory).forEach(builder::addOrReplaceInterceptor);
   }
 
+  private static boolean isHostPresentInNoProxy(String host, String[] noProxyList) {
+    for (String noProxy : noProxyList) {
+      if (isIpAddressRange(noProxy)) {
+        if (new IpAddressMatcher(noProxy).matches(host)) {
+          return true;
+        }
+      } else {
+        if (host.endsWith(noProxy)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private static boolean isValidNoProxy(String[] noProxy) throws MalformedURLException {
+    if (noProxy != null) {
+      for (String np : noProxy) {
+        if (np.indexOf('*') > -1) {
```

Comment:
Could this instead be a check for any invalid url character?  Someone could use other meta/invalid characters.